File: FragmentationX_core/src/main/java/me/liam/support/TransactionRecord.java
package me.liam.support;

import android.os.Bundle;

import java.util.UUID;

import me.liam.anim.FragmentAnimation;
import me.liam.fragmentation.R;

class TransactionRecord {

    FragmentAnimation fragmentAnimation = new FragmentAnimation(R.anim.anim_empty,R.anim.anim_empty,R.anim.anim_empty,R.anim.anim_empty);

    int requestCode = -1;

    int resultCode = -1;

    Bundle resultData;

    boolean displayEnterAnim = true;

    String tag = UUID.randomUUID().toString();

    boolean addBackStack = true;

    Runnable runOnExecute;

    Class popToCls;

    boolean includeTarget = true;

    SupportFragment remove;

    boolean removeAnim = true;

}

File: core/src/main/java/me/liam/fragmentation/support/TransactionRecord.kt
package me.liam.fragmentation.support

import android.os.Bundle

import java.util.UUID

import me.liam.fragmentation.anim.FragmentAnimation
import me.liam.fragmentation.R

internal class TransactionRecord {

    var fragmentAnimation = FragmentAnimation(R.anim.anim_empty, R.anim.anim_empty, R.anim.anim_empty, R.anim.anim_empty)

    var requestCode = -1

    var resultCode = -1

    var resultData: Bundle? = null

    var displayEnterAnim = true

    var tag = UUID.randomUUID().toString()

    var addBackStack = true

    var runOnExecute: Runnable? = null

    var popToCls: Class<*>? = null

    var includeTarget = true

    var remove: SupportFragment? = null

    var removeAnim = true

    var dontDisplaySelfPopAnim = false
}
